### PR TITLE
Iteration 1: add feature to read model definition json

### PR DESCRIPTION
**Pull request by deepakrkris**
add test cases and ability to read a model definition json

#348

### CODE REVIEW

**Comment by ritch** [comment]
We should rename the `workspace-refactor` branch to `4.x`.

File: common/models/model-definition.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+  "properties": {
+    "id": {
+      "type": "string",
+      "id": true,
+      "json": false
+    },
+    "facetName": {
```
Comment (ritch) [inline common/models/model-definition.json]: is this being used yet?

File: common/models/model-definition.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      "type": "string",
+      "required": true,
+      "json": false
+    },
+    "readonly": {
```
Comment (ritch) [inline common/models/model-definition.json]: same comment

File: test/acceptance-tests/workspace-devtime/reading-middleware.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Feature: Users should be able to read middleware configurations
+  As a workspace client using the EXAMPLE workspace
+  I want to query all the middleware configurations in my workspace
+  So that I can reference the middleware phases
+
+  Background: Workspace is loaded in a given directory
+
+  Scenario: get a list of middleware phases
```
Comment (ritch) [inline test/acceptance-tests/workspace-devtime/reading-middleware.feature]: awesome! 👍

File: common/models/model-definition.json (hunk 1 above)
Comment (deepakrkris) [inline common/models/model-definition.json]: No, not yet, but this is required very soon to create rest models for model definitons

**Pull request merged**